Tasks/Maven/maventask.ts

```diff
@@ -68,44 +68,43 @@ var publishJUnitResults = tl.getInput('publishJUnitResults');
 var testResultsFiles = tl.getInput('testResultsFiles', true);
 
 function publishTestResults(publishJUnitResults, testResultsFiles: string) {
+    var matchingTestResultsFiles: string[] = undefined;
     if (publishJUnitResults == 'true') {
         //check for pattern in testResultsFiles
         if (testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
             tl.debug('Pattern found in testResultsFiles parameter');
             var buildFolder = tl.getVariable('agent.buildDirectory');
+            tl.debug(`buildFolder=${buildFolder}`);
             var allFiles = tl.find(buildFolder);
-            var matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, {
+            matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, {
                 matchBase: true
             });
         } else {
             tl.debug('No pattern found in testResultsFiles parameter');
-            var matchingTestResultsFiles = [testResultsFiles];
+            matchingTestResultsFiles = [testResultsFiles];
         }
 
-        if (!matchingTestResultsFiles) {
+        if (!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
             tl.warning('No test result files matching ' + testResultsFiles + ' were found, so publishing JUnit test results is being skipped.');
             return 0;
         }
-
+        
         var tp = new tl.TestPublisher("JUnit");
         tp.publish(matchingTestResultsFiles, true, "", "", "", true);
     }
 }
 
 function getSonarQubeRunner() {
-    var sqAnalysisEnabled = tl.getInput('sqAnalysisEnabled', true);
-
-    if (sqAnalysisEnabled != 'true') {
+    if (!tl.getBoolInput('sqAnalysisEnabled')) {
```

Comment:
No, not required. It seems that we were not very consistent with our use of booleans. I am trying to fix that as I come across them.